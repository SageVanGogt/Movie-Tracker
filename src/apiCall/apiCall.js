import key from './apiKey';

const fetchRecentFilms = async () => {
  const url = `https://api.themoviedb.org/3/discover/movie?api_key=${key}&primary_release_date.gte=2014-09-15&primary_release_date.lte=2014-10-22`;
  try{
    const response = await fetch(url);
    const filmData = await response.json();
    return filmData;
  } catch(err){
    const error = 'Failed to fetch data'
    throw error;
  }
}

const addUserFetch = async (user) => {
  const url = "http://localhost:3000/api/users/new"
  try {
    const response = await fetch(url, {
      method: "POST",
      body: JSON.stringify(user),
      headers: {
        Accept: 'application/json',
        "Content-Type": "application/json"}
    })
    const userData = await response.json();
    return userData;
  } catch(err){
    const error = 'Failed to fetch data'
    throw error;
  }
}

const fetchUser = async (user) => {
  const email = user.email.toLowerCase();
  const  userBody = {
    method: 'POST',
    body: JSON.stringify({ email, password: user.password }),
    headers: {"Content-Type": "application/json"}
  }
  const url = 'http://localhost:3000/api/users';
  try {
    const response = await fetch(url, userBody)
    const userData = await response.json();
    return userData;
  } catch (err) {
    const error = 'Failed to fetch data'
    throw error;
  }
}

const postFavoriteToDb = async (favoriteInfo, user) => {
  const url = "http://localhost:3000/api/users/favorites/new";
  const postData = {
    method: 'POST',
    body: JSON.stringify({user_id: user.user_id, ...favoriteInfo}),
    headers: {
      Accept: 'application/json',
      "Content-Type": "application/json"
    }
  }
  try {
    await fetch(url, postData);
  } catch(err) {
    const error = "Failed to post favorite";
    throw error;
  }
}

const getUserFavorites = async (userId) => {
  const url = `http://localhost:3000/api/users/${userId}/favorites`;
  try{
  const response = await fetch(url);
  const favorites = await response.json();
  return favorites;
  } catch(err) {
    const error = "Failed to get favorites";
    throw error;
  }
}

const removeFavorite = async (favorite) => {
  debugger
  const url = `http://localhost:3000/api/users/${favorite.user_id}/favorites/${favorite.movie_id}`
  try{
    const response = await fetch(url, {
      method: 'DELETE',
      headers: {
        "Content-Type": "application/json",
        },
      body: JSON.stringify(favorite)

    })
    const data = await response.json();
  } catch(err) {
    const error = "Failed to remove favorite";
    throw error;
  }
}


export {
  fetchRecentFilms, 
  addUserFetch, 
  fetchUser,
  postFavoriteToDb,
  getUserFavorites,
  removeFavorite
};