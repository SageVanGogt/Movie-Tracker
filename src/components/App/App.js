import React, { Component } from 'react';
import fetchFilmData from './../../apiCall/apiCall';
import cleanFilmData from './../../helper/helper';
import * as actions from './../../actions/index';
import './App.css';

class App extends Component {
  
  async componentDidMount() {
    const recentFilms = await fetchFilmData();
    const pageLoadFilms = cleanFilmData(recentFilms);
    this.props.addRecentFilms(pageLoadFilms);
  }

  render() {
    return (
      <div className="App">
       HI
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  addRecentFilms: (pageLoadFilms) => dispatch(action.addRecentFilms(pageLoadFilms))
}

export default App;
