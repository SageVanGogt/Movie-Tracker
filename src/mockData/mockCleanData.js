const mockCleanData = [
          {
            "movie_id": 245891,
            "vote_average": 7,
            "title": "John Wick",
            "poster_path": "https://image.tmdb.org/t/p/w500/5vHssUeVe25bMrof1HyaPyWgaP.jpg",
            "overview": "Ex-hitman John Wick comes out of retirement to track down the gangsters that took everything from him.",
            "release_date": "2014-10-22"
          },
          {
            "movie_id": 228150,
            "vote_average": 7.4,
            "title": "Fury",
            "poster_path": "https://image.tmdb.org/t/p/w500/pfte7wdMobMF4CVHuOxyu6oqeeA.jpg",
            "overview": "Last months of World War II in April 1945. As the Allies make their final push in the European Theater, a battle-hardened U.S. Army sergeant in the 2nd Armored Division named Wardaddy commands a Sherman tank called \"Fury\" and its five-man crew on a deadly mission behind enemy lines. Outnumbered and outgunned, Wardaddy and his men face overwhelming odds in their heroic attempts to strike at the heart of Nazi Germany.",
            "release_date": "2014-10-15"
          },
          {
            "movie_id": 210577,
            "vote_average": 7.9,
            "title": "Gone Girl",
            "poster_path": "https://image.tmdb.org/t/p/w500/gdiLTof3rbPDAmPaCf4g6op46bj.jpg",
            "overview": "With his wife's disappearance having become the focus of an intense media circus, a man sees the spotlight turned on him when it's suspected that he may not be innocent.",
            "release_date": "2014-10-01"
          },
          {
            "movie_id": 156022,
            "vote_average": 7.1,
            "title": "The Equalizer",
            "poster_path": "https://image.tmdb.org/t/p/w500/2eQfjqlvPAxd9aLDs8DvsKLnfed.jpg",
            "overview": "In The Equalizer, Denzel Washington plays McCall, a man who believes he has put his mysterious past behind him and dedicated himself to beginning a new, quiet life. But when McCall meets Teri (Chloë Grace Moretz), a young girl under the control of ultra-violent Russian gangsters, he can’t stand idly by – he has to help her. Armed with hidden skills that allow him to serve vengeance against anyone who would brutalize the helpless, McCall comes out of his self-imposed retirement and finds his desire for justice reawakened. If someone has a problem, if the odds are stacked against them, if they have nowhere else to turn, McCall will help. He is The Equalizer.",
            "release_date": "2014-09-24"
          }]

export default mockCleanData;