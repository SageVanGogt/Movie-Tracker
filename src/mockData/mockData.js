const mockData = {
  "page": 1,
  "total_results": 1801,
  "total_pages": 91,
  "results": [
    {
      "vote_count": 7464,
      "id": 245891,
      "video": false,
      "vote_average": 7,
      "title": "John Wick",
      "popularity": 46.227181,
      "poster_path": "/5vHssUeVe25bMrof1HyaPyWgaP.jpg",
      "original_language": "en",
      "original_title": "John Wick",
      "genre_ids": [
        28,
        53
      ],
      "backdrop_path": "/umC04Cozevu8nn3JTDJ1pc7PVTn.jpg",
      "adult": false,
      "overview": "Ex-hitman John Wick comes out of retirement to track down the gangsters that took everything from him.",
      "release_date": "2014-10-22"
    },
    {
      "vote_count": 5107,
      "id": 228150,
      "video": false,
      "vote_average": 7.4,
      "title": "Fury",
      "popularity": 41.861592,
      "poster_path": "/pfte7wdMobMF4CVHuOxyu6oqeeA.jpg",
      "original_language": "en",
      "original_title": "Fury",
      "genre_ids": [
        10752,
        18,
        28
      ],
      "backdrop_path": "/pKawqrtCBMmxarft7o1LbEynys7.jpg",
      "adult": false,
      "overview": "Last months of World War II in April 1945. As the Allies make their final push in the European Theater, a battle-hardened U.S. Army sergeant in the 2nd Armored Division named Wardaddy commands a Sherman tank called \"Fury\" and its five-man crew on a deadly mission behind enemy lines. Outnumbered and outgunned, Wardaddy and his men face overwhelming odds in their heroic attempts to strike at the heart of Nazi Germany.",
      "release_date": "2014-10-15"
    },
    {
      "vote_count": 7676,
      "id": 210577,
      "video": false,
      "vote_average": 7.9,
      "title": "Gone Girl",
      "popularity": 41.666242,
      "poster_path": "/gdiLTof3rbPDAmPaCf4g6op46bj.jpg",
      "original_language": "en",
      "original_title": "Gone Girl",
      "genre_ids": [
        9648,
        53,
        18
      ],
      "backdrop_path": "/bt6DhdALyhf90gReozoQ0y3R3vZ.jpg",
      "adult": false,
      "overview": "With his wife's disappearance having become the focus of an intense media circus, a man sees the spotlight turned on him when it's suspected that he may not be innocent.",
      "release_date": "2014-10-01"
    },
    {
      "vote_count": 3606,
      "id": 156022,
      "video": false,
      "vote_average": 7.1,
      "title": "The Equalizer",
      "popularity": 35.007408,
      "poster_path": "/2eQfjqlvPAxd9aLDs8DvsKLnfed.jpg",
      "original_language": "en",
      "original_title": "The Equalizer",
      "genre_ids": [
        53,
        28,
        80
      ],
      "backdrop_path": "/ms9oACSGJRjofNPA6HVtcgE5DiU.jpg",
      "adult": false,
      "overview": "In The Equalizer, Denzel Washington plays McCall, a man who believes he has put his mysterious past behind him and dedicated himself to beginning a new, quiet life. But when McCall meets Teri (Chloë Grace Moretz), a young girl under the control of ultra-violent Russian gangsters, he can’t stand idly by – he has to help her. Armed with hidden skills that allow him to serve vengeance against anyone who would brutalize the helpless, McCall comes out of his self-imposed retirement and finds his desire for justice reawakened. If someone has a problem, if the odds are stacked against them, if they have nowhere else to turn, McCall will help. He is The Equalizer.",
      "release_date": "2014-09-24"
    } 
  ] 
};

export default mockData;
