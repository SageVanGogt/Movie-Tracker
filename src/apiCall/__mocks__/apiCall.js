import mockData from './../../mockData/mockData';
const fetchRecentFilms = jest.fn().mockImplementation(() => Promise.resolve(mockData))
const addUserFetch = jest.fn().mockImplementation(() => Promise.resolve({
    name: "",
    email: "great",
    password: ""
}))

export {fetchRecentFilms, addUserFetch};

