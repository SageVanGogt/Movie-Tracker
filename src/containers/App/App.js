import React, { Component } from 'react';
import { fetchRecentFilms, getUserFavorites } from './../../apiCall/apiCall';
import cleanFilmData from './../../helper/helper';
import { connect } from 'react-redux';
import { addRecentFilms, addFavoritesToStore } from './../../actions/index';
import { Route, Redirect, Switch, withRouter } from 'react-router-dom';
import './App.css';
import Nav from '../Nav/Nav';
import Login from '../../components/Login/Login';
import RecentMovies from '../RecentMovies/RecentMovies';
import Favorites from '../Favorites/Favorites';

export class App extends Component {
  
  componentDidMount() {
    this.getRecentFilms()
  }

  async getRecentFilms() {
    const recentFilms = await fetchRecentFilms();
    const pageLoadFilms = cleanFilmData(recentFilms.results);
    this.props.handlePageLoadFilms(pageLoadFilms);
  }

  getFavorites = async () => {
    if (this.props.user.name) {
      const favorites = await getUserFavorites(this.props.user.user_id);
      this.props.populateFavorites(favorites.data)
    }
  }

  render() {
    {this.getFavorites()}
    return (
      <div className="App">
        <header> 
          <Nav />
        </header>
        <Switch>
          <Route 
            exact path="/login" 
            render= {() => (
              this.props.user.user_id ?
              <Redirect to="/" /> :
              <Login />
            )}
          />
          <Route 
            exact path="/" 
            component={RecentMovies}
          />
          <Route 
            exact path="/favorites"
            component={Favorites}/>
        </Switch>
      </div> 
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  handlePageLoadFilms: (pageLoadFilms) => dispatch(addRecentFilms(pageLoadFilms)),
  populateFavorites: (movies) => dispatch(addFavoritesToStore(movies))
})

export const mapStateToProps = (state) => ({
  user: state.user
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
