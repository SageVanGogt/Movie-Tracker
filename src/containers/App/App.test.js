import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import mockCleanData from './../../mockData/mockCleanData';
import mockData from './../../mockData/mockData';
import { fetchRecentFilms } from './../../apiCall/apiCall';

jest.mock('./../../apiCall/apiCall');
describe('App', () => {
  let wrapper;
  let mockHandlePageLoadFilms;
  let mockHandleError;
  let mockProps;
  
  beforeEach(() => {
    mockHandlePageLoadFilms = jest.fn();
    mockHandleError = jest.fn()
    mockProps = {
      error: ''
    }

    wrapper = shallow(<App
      handlePageLoadFilms={mockHandlePageLoadFilms}
      handleError={mockHandleError}
      {...mockProps}
    />, { disableLifeCycleMethods: true });
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('getRecentFilms should call fetchRecentFilms on pageload', async () => {
    await wrapper.instance().getRecentFilms();

    expect(fetchRecentFilms).toHaveBeenCalled()
  })

  it.skip('getRecentFilms should call cleanFilmData on with fetchedFilms', async () => {
    let cleanFilmData = jest.fn()
    await wrapper.instance().getRecentFilms();

    expect(cleanFilmData).toHaveBeenCalledWith()
  })

  it('should call handlePageLoadFilms with the correct params', async () => {
    await wrapper.instance().getRecentFilms()

    expect(mockHandlePageLoadFilms).toHaveBeenCalledWith(mockCleanData)
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct params', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'ADD_RECENT',
        films: mockCleanData
      }
      mappedProps.handlePageLoadFilms(mockCleanData);

      expect(mockDispatch).toHaveBeenCalledWith(mockAction)
    })

    it('should call dispatch on handleError with the correct params', () => {
      const mockDispatch = jest.fn()
      const mappedProps = mapDispatchToProps(mockDispatch)
      const mockAction = {
        type: "ADD_ERROR",
        message: "Sign in to add favorites"
      }
      const mockError = "Sign in to add favorites";

      mappedProps.handleError(mockError)

      expect(mockDispatch).toHaveBeenCalledWith(mockAction)
    });
  })

  describe('mapStateToProps', () => {
    it('should map the user to props', () => {
      const mockState = {
        user: {
          id: 1,
          name: 'Alan',
          },
        films: []
      }

      const expected = {
        user: {
          id: 1,
          name: 'Alan',
        }
      }
      
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
})



