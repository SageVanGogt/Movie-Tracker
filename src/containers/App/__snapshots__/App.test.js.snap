// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    handlePageLoadFilms={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <Connect(Component) />
</header>,
        <Switch>
          <Route
                    exact={true}
                    path="/login"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/favorites"
          />
</Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Component) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/login"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/favorites"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/login",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/favorites",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <Connect(Component) />
</header>,
          <Switch>
            <Route
                        exact={true}
                        path="/login"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/favorites"
            />
</Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Component) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/login"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/favorites"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/login",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/favorites",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifeCycleMethods": true,
  },
}
`;
